test/unit/math/mathOps/include/DataGen.hpp

```diff
@@ -0,0 +1,139 @@
+/** Copyright 2019 Jakob Krude, Benjamin Worpitz
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include "Defines.hpp"
+#include <random>
+#include <limits>
+#include <cassert>
+
+namespace alpaka {
+namespace test {
+namespace unit {
+namespace math {
+
+
+    /**
+     * Fills buffer with random numbers (host-only).
+     *
+     * @tparam TData The used data-type (float || double).
+     * @tparam TArgs The args-buffer to be filled.
+     * @tparam TFunctor The used Functor-type.
+     * @param args The buffer that should be filled.
+     * @param functor The Functor, needed for ranges.
+     * @param seed The used seed.
+     */
+    template<
+        typename TData,
+        typename TArgs,
+        typename TFunctor>
+    auto fillWithRndArgs(
+        TArgs  & args,
+        TFunctor functor,
+        unsigned long const & seed
+    ) -> void
+    {
+        /*
+         * Each "sub-buffer" is filled with zero and/or max and/or lowest,
+         * depending on the specified range (at [0] - [2]).
+         *
+         * Every switch case needs to return!
+         * If no switch case was matched an assert(false) will be triggered.
+         *
+         * This function is easily extendable. It is only necessary to add extra
+         * definitions in the switch case, for more Range-types.
+         */
+        static_assert( TArgs::value_type::arity == TFunctor::arity,
+            "Buffer properties must match TFunctor::arity" );
+        static_assert( TArgs::capacity > 2,
+            "Set of args must provide > 2 entries." );
+        constexpr auto max = std::numeric_limits< TData >::max();
+        constexpr auto low = std::numeric_limits< TData >::lowest();
+        std::default_random_engine eng{ seed };
```

Comment:
You get a cast warning here:
`error: non-constant-expression cannot be narrowed from type 'unsigned long' to 'std::__1::linear_congruential_engine<unsigned int, 48271, 0, 2147483647>::result_type' (aka 'unsigned int') in initializer list [-Wc++11-narrowing]`
This can probably be fixed by:
```suggestion
        std::default_random_engine eng{ static_cast<std::default_random_engine::result_type>(seed) };
```